optuna/integration/pytorch_distributed.py

```diff
@@ -0,0 +1,249 @@
+from datetime import datetime
+import pickle
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+import warnings
+
+import optuna
+from optuna._experimental import experimental
+from optuna._imports import try_import
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalChoiceType
+
+
+with try_import() as _imports:
+    import torch
+    import torch.distributed as dist
+
+
+@experimental("2.6.0")
+class TorchDistributedTrial(optuna.trial.BaseTrial):
+    """A wrapper of :class:`~optuna.trial.Trial` to incorporate Optuna with PyTorch distributed.
+
+    .. seealso::
+        :class:`~optuna.integration.TorchDistributedTrial` provides the same interface as
+        :class:`~optuna.trial.Trial`. Please refer to :class:`optuna.trial.Trial` for further
+        details.
+
+    See `the example <https://github.com/optuna/optuna/blob/master/
+    examples/pytorch_distributed_simple.py>`__
+    if you want to optimize an objective function that trains neural network
+    written with PyTorch distributed data parallel.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` object or :obj:`None`. Please set trial object in
+            rank-0 node and set :obj:`None` in the other rank node.
+    """
+
+    def __init__(self, trial: Optional[optuna.trial.Trial]) -> None:
+
+        _imports.check()
```

Comment:
I'm not sure it's acceptable, but how about adding a check like the following to detect wrong usage of this class earlier?
```suggestion
        _imports.check()
        
        if dist.get_rank() == 0:
            assert trial is not None, "error message"
        else:
            assert trial is None, "error message"
```

Reply:
Thank you for your suggestion. That makes sense. I think we can raise `ValueError` for the invalid input values, and I changed the error type in the commit https://github.com/optuna/optuna/pull/2303/commits/2d49ceb88dcce930f898f074d1beeea6d1af19e6.